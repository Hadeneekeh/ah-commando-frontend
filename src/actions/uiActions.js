import {
  MODAL_OPEN,
  MODAL_CLOSE,
} from '@Actions/types';

export const openModal = modal => ({
  type: MODAL_OPEN,
  payload: modal,
});

export const closeModal = () => ({
  type: MODAL_CLOSE,
});
