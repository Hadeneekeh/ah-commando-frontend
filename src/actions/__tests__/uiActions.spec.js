import { openModal, closeModal } from '@Actions/uiActions';

describe('UI Action tests', () => {
  const payload = 'signin';
  it('openModal should return MODAL_OPEN type with the right modal as payload', () => {
    expect(openModal(payload)).toMatchSnapshot();
  });

  it('closeModal should return MODAL_CLOSE type', () => {
    expect(closeModal(payload)).toMatchSnapshot();
  });
});
