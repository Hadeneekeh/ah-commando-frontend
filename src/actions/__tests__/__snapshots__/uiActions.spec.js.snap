// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Action tests closeModal should return MODAL_CLOSE type 1`] = `
Object {
  "type": "MODAL_CLOSE",
}
`;

exports[`UI Action tests openModal should return MODAL_OPEN type with the right modal as payload 1`] = `
Object {
  "payload": "signin",
  "type": "MODAL_OPEN",
}
`;
